theano/tests/test_gradient.py

```diff
@@ -642,7 +641,110 @@ def test_grad(self):
 
             assert np.allclose(f(a), f2(a))
 
+class TestZeroGrad(unittest.TestCase):
+
+    def setUp(self):
+        utt.seed_rng()
+        self.rng = np.random.RandomState(seed=utt.fetch_seed())
+
+    def test_op_removed(self):
+        x = theano.tensor.matrix('x')
+        y = x * gradient.zero_grad(x)
+        f = theano.function([x], y)
+        # need to refer to theano.gradient.zero_grad here,
+        # theano.gradient.zero_grad is a wrapper function!
+        assert gradient.zero_grad_ not in \
+            [node.op for node in f.maker.fgraph.toposort()]
+        
+    def test_grad(self):
+        T = theano.tensor
+        a = np.asarray(self.rng.randn(5, 5),
+            dtype=config.floatX)
+        
+        x = T.matrix('x')
+
+        expressions_gradients = [
+            (x * gradient.zero_grad(x), x),
+            (x * gradient.zero_grad(T.exp(x)), T.exp(x)),
+            (gradient.zero_grad(x), T.constant(0.)),
+            (x**2 * gradient.zero_grad(x), 2 * x**2),
+        ]
+
+        for expr, expr_grad in expressions_gradients:
+            g = gradient.grad(expr.sum(), x)
+            # gradient according to theano
+            f = theano.function([x], g, on_unused_input='ignore')
+            # desired gradient
+            f2 = theano.function([x], expr_grad, on_unused_input='ignore')
+
+            assert np.allclose(f(a), f2(a))
+
+def grad_wrt_expression(exp, grad_wrt_variable):
+    return gradient.grad(exp, grad_wrt_variable)
```

Comment:
Nothing actually, it was a function I used earlier but now got rid of it... I'll remove it :)